Feat: csv feature added

TitanX311 opened the pull request:
Fixed issue #10 
local testing and state persistance done

TitanX311 committed:
csv feature added

xenonnn4w commented:
Where is the description of your fix? Where is the issue number? Did you tested the feature locally?

TitanX311 committed:
reverted #8

xenonnn4w commented:
Rebase it to upstream @TitanX311

xenonnn4w commented:
@TitanX311  rebase it to upstream; remove the merge conflicts.

TitanX311 committed:
Merge branch 'main' into fix-issue-10

TitanX311 commented:
### Summary
- Resolved all merge conflicts
- Added CSV export for assignments and grades
- Fixed datetime serialization with custom JSON encoder/decoder

### Tested
- add-assignment
- add-grade
- list
- export assignments / grades

Ready to merge
